BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -183,12 +185,29 @@ public actor DefaultTabDataStore: TabDataStore {
                     }
                 }
             }
-            try await self.writeWindowData(windowData: window, to: windowSavingPath)
+            try await self.throttle {
+                try await self.writeWindowData(windowData: window, to: windowSavingPath)
+            }
         } catch {
             self.logger.log("Failed to save window data: \(error)", level: .debug, category: .tabs)
         }
     }
 
+    func throttle(operation: @escaping () async throws -> Void) async throws {
+        guard saveTask == nil else { return }
+
+        saveTask = Task {
+            defer { saveTask = nil }
+            try await Task.sleep(nanoseconds: UInt64(throttleInterval))
+        }
+
+        do {
+            try await operation()
+        } catch {
+            throw error
+        }
+    }
```

Comment:
I don't think this is doing what you think it is doing. The saveTask block is independent of the section that calls the operation function, so it isn't blocking that call at all. 
You can also tell this isn't working right because the test currently passes even tho it doesn't wait 5 seconds.